android/java/org/chromium/chrome/browser/crypto_wallet/activities/AddAccountActivity.java

```diff
@@ -233,16 +222,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
     @Override
     public void finishNativeInitialization() {
         super.finishNativeInitialization();
+        if (getIntent() != null) {
+            mAddress = getIntent().getStringExtra(Utils.ADDRESS);
+            mName = getIntent().getStringExtra(Utils.NAME);
+            mIsImported = getIntent().getBooleanExtra(Utils.ISIMPORTED, false);
+        }
+        if (!TextUtils.isEmpty(mAddress) && !TextUtils.isEmpty(mName)) {
```

Comment:
I think it is better send a boolean from Intent rather than relying on this condition which seems unreliable.

Reply:
sounds good, I updated it to pass the boolean via Intent